eng/scripts/Get-TypeSpec-Folders.ps1

```diff
@@ -1,61 +1,42 @@
 [CmdletBinding()]
 param (
-  [Parameter(Position = 0, Mandatory = $true)]
-  [string] $SpecsRepoRootDirectory,
-  [Parameter(Position = 1, Mandatory = $false)]
-  [string]$TargetBranch,
-  [Parameter(Position = 2, Mandatory = $false)]
-  [string]$SourceBranch
+  [switch]$CheckAll = $false
 )
+Set-StrictMode -Version 3
 
-$changedFiles = @()
-$allChangedFiles = (Get-ChildItem -path ./specification tspconfig.* -Recurse).Directory.FullName | ForEach-Object {[IO.Path]::GetRelativePath($($pwd.path), $_)}
-$allChangedFiles = $allChangedFiles -replace '\\', '/'
+. $PSScriptRoot/ChangedFiles-Functions.ps1
 
-if ([string]::IsNullOrEmpty($TargetBranch) -or [string]::IsNullOrEmpty($SourceBranch)) {
-  if ($TargetBranch -or $SourceBranch) {
-    throw "Please provide both target branch and source branch."
-  }
-  $changedFiles = $allChangedFiles
+$repoPath = Resolve-Path "$PSScriptRoot/../.."
+$checkAllPath = "specification"
+
+if ($CheckAll) {
+  $changedFiles = $checkAllPath
 }
 else {
-  Write-Host "git -c core.quotepath=off -c i18n.logoutputencoding=utf-8 diff --name-only `"$TargetBranch...$SourceBranch`" --"
-  $changedFiles = git -c core.quotepath=off -c i18n.logoutputencoding=utf-8 diff --name-only `"$TargetBranch...$SourceBranch`" --
-  $changedFiles = $changedFiles -replace '\\', '/' | Sort-Object
+  $changedFiles = Get-ChangedFiles
```

Comment:
```suggestion
  $changedFiles = Get-ChangedFiles "HEAD^" "HEAD" ""
```

`Get-TypeSpec-Folders.ps1` needs to use deleted files as a signal to validate specs.